packages/bundler-plugin-core/src/utils/getPreSignedURL.ts

```diff
@@ -55,6 +63,29 @@ export const getPreSignedURL = async ({
 
       requestBody.git_service = foundGitService;
     }
+  } else if (oidc?.useGitHubOIDC && Core) {
+    if (serviceParams?.service !== "github-actions") {
+      red("OIDC is only supported for GitHub Actions");
+      throw new BadOIDCServiceError(
+        "OIDC is only supported for GitHub Actions",
+      );
+    }
+
+    let token = "";
+    try {
+      token = await Core.getIDToken(oidc.gitHubOIDCTokenAudience);
+    } catch (err) {
+      if (err instanceof Error) {
+        red(
+          `Failed to get OIDC token with url:\`${oidc.gitHubOIDCTokenAudience}\`. ${err.message}`,
+        );
+        throw new FailedOIDCFetchError(
```

Comment:
we should attach `err` as the `error.cause` of the `FailedOIDCFetchError`

Reply:
Hrm, I'm getting type errors trying to add a second argument to `Error`, I see it in the MDN docs. I tried updating the `@types/node` however that didn't resolve the issue, any other ideas/guidance 👀

Reply:
Hmm it might be because of the tsconfig here: https://github.com/codecov/codecov-javascript-bundler-plugins/blob/be89828f37952cca8141cea1cfc24dc8ab1fa9d8/tsconfig.json#L11

try setting it to be ES2022?

if that doesn't work feel free to ignore this and more forward.

Reply:
ahh yea that did the trick